doc-content/kogito-docs/src/main/asciidoc/configuration/con_task-life-cycle.adoc

```diff
@@ -0,0 +1,119 @@
+* *Release*: Unassigns the given work item from a specific actor, releasing it to any other potential user or group to work on it (by claiming or completing)
+* *Skip*: Skips the given work item
+
+With {PRODUCT}, you can also add custom life cycles and life cycle phases to meet your business needs.
+
+A life cycle moves a work item across various phases that are not defined by the `WorkItem` interface and defines the behavior of a given work item at runtime. You typically add a life cycle on top of the `WorkItemHandler` interface to so that the life cycle is pluggable with more flexible runtime characteristics.
+
+The `WorkItemHandler` interface provides the option to move between task phases, as shown in the following method example:
+
+.WorkItemHandler support for moving between task phases
+[source, java]
+----
+public void transitionToPhase(WorkItem workItem, WorkItemManager manager, Transition<?> transition)
+----
+
+NOTE: This method is a default method that does nothing when not implemented. This functionality maintains backward compatibility with existing work item handler implementations.
+
+You typically implement the `transitionToPhase` method as shown in the following example:
+
+.Example implementation of transitionToPhase
+[source, java]
+----
+@Override
+public void transitionToPhase(WorkItem workItem, WorkItemManager manager, Transition<?> transition) {
+
+    lifeCycle.transitionTo(workItem, manager, (Transition<Map<String, Object>>) transition);
+}
+----
+
+The `lifeCycle` element is an implementation of `org.kie.{PRODUCT_INIT}.process.workitem.LifeCycle<T>` that defines the execution semantics.
+
+== Task authorization
+
+The `org.jbpm.process.instance.impl.humantask.BaseHumanTaskLifeCycle` implementation in {PRODUCT} ensures that a user task is worked on by authorized users, based on the user or group assignments that you provide.
+
+You can use the following parameters to provide assignments for authorized users or groups in the relevant BPMN process model. All of the listed paramters support expressions.
+
+.Parameters for authorized users or groups
+[cols="40%,40%,20%"]
+|===
+|Parameter name |Description |Example value
+
+|`ActorId`
+|Comma-separated list of authorized users
+|John,Mary,#{actor}
+
+|`GroupId`
+|Comma-separated list of authorized groups of users
+|mangers,#{mygroup}
+
+|`BusinessAdministratorId`
+|Comma-separated list of authorized administrators
+|Administrator,#{adminuser}
+
+|`BusinessAdministratorGroupId`
+|Comma-separated list of groups of administrators
+|admins,#{admingroup}
+
+|`ExcludedOwnerId`
+|Comma-separated list of unauthorized users who cannot work on this task
+|poul,#{lastactor}
+|===
+
+NOTE: Authorization is only enforced when the method that calls the work item life cycle methods comes with a security context. This security behavior is dependent on the API that you use.
+
+== API interaction with task life cycle phases
```

Comment:
@sterobin we should at least mention the REST endpoints here as another way to interact with the tasks and show the endpoint that are created. I belive you covered the content in some examples already.

Reply:
I see. I don't think I get into any example endpoints elsewhere specifically for tasks, but this is my attempt using one of the [orderItems.bpmn2](https://github.com/kiegroup/kogito-examples/blob/stable/jbpm-quarkus-example/src/main/resources/org/kie/kogito/examples/orderItems.bpmn) supported endpoints, which I've added to the end of this section. 

**However, I can't get this to work, so clearly something is wrong. Can you see what I'm missing?**

```
The following REST endpoint example sets a task phase, group, and user in an `orderItems.bpmn2` process for verifying customer orders:

.Example POST endpoint to set a task phase, group, and user
[source]
----
http://localhost:8080/orderItems/__PROCESS_ID__/Verify_order/__WORK_ITEM_ID__?phase=__PHASE_NAME__&group=__GROUP_NAMES__&user=__USER_NAME__
----

.Example curl command to claim a task for a specified group and user
[source]
----
curl -X POST "http://localhost:8080/orderItems/demo.orderItems/Verify_order/UserTask_1?phase=claim&group=managers&user=john" -H "accept: application/json" -H "content-type: application/json"
----

.Example curl command to complete a task for a specified group and user
[source]
----
curl -X POST "http://localhost:8080/orderItems/demo.orderItems/Verify_order/UserTask_1?phase=complete&group=managers&user=john" -H "accept: application/json" -H "content-type: application/json"
----
```

Reply:
@sterobin should be a lot simpler than that, see https://github.com/kiegroup/kogito-examples/tree/stable/jbpm-quarkus-example#get-orderitemsidtasks.
Once you know the process id, you only need to do a get to retrieve the open tasks and a post to complete the task using both process id and task id as parameter.